packages/cloud_functions/cloud_functions/macos/cloud_functions/Package.swift

```diff
@@ -0,0 +1,117 @@
+// swift-tools-version: 5.9
+// The swift-tools-version declares the minimum version of Swift required to build this package.
+
+// Copyright 2024, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import Foundation
+import PackageDescription
+
+enum ConfigurationError: Error {
+  case fileNotFound(String)
+  case parsingError(String)
+  case invalidFormat(String)
+}
+
+let functionsDirectory = String(URL(string: #file)!.deletingLastPathComponent().absoluteString
+  .dropLast())
+
+func loadFirebaseSDKVersion() throws -> String {
+  let firebaseCoreScriptPath = NSString.path(withComponents: [
+    functionsDirectory,
+    "..",
+    "..",
+    "ios",
+    "generated_firebase_sdk_version.txt",
+  ])
+  do {
+    let version = try String(contentsOfFile: firebaseCoreScriptPath, encoding: .utf8)
+      .trimmingCharacters(in: .whitespacesAndNewlines)
+    return version
+  } catch {
+    throw ConfigurationError
+      .fileNotFound("Error loading or parsing generated_firebase_sdk_version.txt: \(error)")
+  }
+}
+
```

Comment:
Do you think this could be shared between different plugins? Or is this generated for each one?

Reply:
It's not possible in a clean way. Can't import local files into Package.swift and can't specify a dependency to be used in Package.swift 😔